### PR TITLE
Website - Standardize and improve how we document "named blocks", yielded content, and contextual components

### PR DESCRIPTION
**Pull request by didoo**
### :pushpin: Summary

This is an attempt to bring some standardization (and in the process improvement where necessary) on how we document 
- named blocks
- yielded content
- contextual component

This attempt has been sparkled by @MelSumner (see [thread here](https://hashicorp.slack.com/archives/C025N5V4PFZ/p1720461683681729)) in the context of #1945 (see [preview here](https://hds-website-git-dialog-primitive-non-breaking-58ad01-hashicorp.vercel.app/components/flyout?tab=code#contextual-components))

### :hammer_and_wrench: Detailed description

In this PR I have extensively changed/reworked the documentation in the "Component API" blocks for a lot of components (see list below). I haven't touched yet the form-related components (I'll do it once we converge/decide if the format proposed in this PR is OK)

### 👀 Previews:

- 👉 [Accordion](https://hds-website-git-website-documentation-contextu-7c291f-hashicorp.vercel.app/components/accordion?tab=code#component-api)
- 👉 [Alert](https://hds-website-git-website-documentation-contextu-7c291f-hashicorp.vercel.app/components/alert?tab=code#component-api)
- 👉 [App Footer](https://hds-website-git-website-documentation-contextu-7c291f-hashicorp.vercel.app/components/app-footer?tab=code#component-api)
- 👉 [Application State](https://hds-website-git-website-documentation-contextu-7c291f-hashicorp.vercel.app/components/application-state?tab=code#component-api)
- 👉 [Code Block](https://hds-website-git-website-documentation-contextu-7c291f-hashicorp.vercel.app/components/code-block?tab=code#component-api)
- 👉 [Dropdown](https://hds-website-git-website-documentation-contextu-7c291f-hashicorp.vercel.app/components/dropdown?tab=code#component-api)
- 👉 [Flyout](https://hds-website-git-website-documentation-contextu-7c291f-hashicorp.vercel.app/components/flyout?tab=code#component-api)
- 👉 [Modal](https://hds-website-git-website-documentation-contextu-7c291f-hashicorp.vercel.app/components/modal?tab=code#component-api)
- 👉 [Page Header](https://hds-website-git-website-documentation-contextu-7c291f-hashicorp.vercel.app/components/page-header?tab=code#component-api)
- 👉 [Rich Tooltip](https://hds-website-git-website-documentation-contextu-7c291f-hashicorp.vercel.app/components/rich-tooltip?tab=code#component-api)
- 👉 [Segmented Group](https://hds-website-git-website-documentation-contextu-7c291f-hashicorp.vercel.app/components/segmented-group?tab=code#component-api)
- 👉 [Separator](https://hds-website-git-website-documentation-contextu-7c291f-hashicorp.vercel.app/components/separator?tab=code#component-api)
- 👉 [Side Nav](https://hds-website-git-website-documentation-contextu-7c291f-hashicorp.vercel.app/components/side-nav?tab=code#component-api)
- 👉 [Tabs](https://hds-website-git-website-documentation-contextu-7c291f-hashicorp.vercel.app/components/tabs?tab=code#component-api)
- 👉 [AppFrame](https://hds-website-git-website-documentation-contextu-7c291f-hashicorp.vercel.app/layouts/app-frame?tab=code#component-api)

### :link: External links

Jira ticket: https://hashicorp.atlassian.net/browse/HDS-3549

***

### 👀 Component checklist

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: website/docs/components/dropdown/partials/code/component-api.md [hunk 1]
```diff
@@ -97,7 +128,63 @@ If the Dropdown content exceeds the height of the container, the header and foot
...
+  <C.Property @name="...attributes">
+    This component supports use of [`...attributes`](https://guides.emberjs.com/release/in-depth-topics/patterns-for-components/#toc_attribute-ordering).
+  </C.Property>
+</Doc::ComponentApi>
+
+#### [DD].Interactive
+
+The `Dropdown::ListItem::Interactive` component, yielded as contextual component.
+
+It internally uses the [`Hds::Interactive`](/utilities/interactive) utility component. For more details about this component API please refer to [its documentation page](/utilities/interactive?tab=code#component-api).
```
Comment (KristinLBradley) [inline website/docs/components/dropdown/partials/code/component-api.md:187]:
```suggestion
It internally uses the [`Hds::Interactive`](/utilities/interactive) utility component. For more details about this component API, please refer to [its documentation page](/utilities/interactive?tab=code#component-api).
```

File: website/docs/components/dropdown/partials/code/component-api.md [hunk 2]
```diff
@@ -134,37 +221,11 @@ If the Dropdown content exceeds the height of the container, the header and foot
...
 
-<Doc::ComponentApi as |C|>
-  <C.Property @name="...attributes">
-    This component supports use of [`...attributes`](https://guides.emberjs.com/release/in-depth-topics/patterns-for-components/#toc_attribute-ordering).
-  </C.Property>
-</Doc::ComponentApi>
+The `Dropdown::ListItem::CopyItem` component, yielded as contextual component.
 
-### ListItem::CopyItem
+It internally uses the [`Copy::Snippet`](/components/copy/snippet) component. For more details about this component API please refer to [its documentation page](/components/copy/snippet?tab=code#component-api).
```
Comment (KristinLBradley) [inline website/docs/components/dropdown/partials/code/component-api.md:228]:
```suggestion
It internally uses the [`Copy::Snippet`](/components/copy/snippet) component. For more details about this component API, please refer to [its documentation page](/components/copy/snippet?tab=code#component-api).
```

File: website/docs/components/app-footer/partials/code/component-api.md [hunk 3]
```diff
@@ -6,22 +6,22 @@ The base `AppFooter` component includes a copyright notice. It also wraps and co
 
 <Doc::ComponentApi as |C|>
   <C.Property @name="<[AF].ExtraBefore>" @type="yielded component">
-    Container that yields its content before the `<ul>` list of links and items.
+    A generic container yielded as contextual component. Its content is injected before the `<ul>` list of links and items.
   </C.Property>
   <C.Property @name="<[AF].StatusLink>" @type="yielded component">
-    The `AppFooter::StatusLink` component (see below).
+    The yielded `AppFooter::StatusLink` contextual component ([see below](#appfooterstatuslink)).
```
Comment (alex-ju) [inline website/docs/components/app-footer/partials/code/component-api.md:12]: I don't believe these links go anywhere. should we make them text only as in other components?

File: website/docs/components/dropdown/partials/code/component-api.md [hunk 4]
```diff
@@ -17,6 +17,45 @@ The Dropdown component is composed of different child components each with their
 ### Dropdown
 
 <Doc::ComponentApi as |C|>
+  <C.Property @name="<[DD].ToggleButton>" @type="yielded component">
```
Comment (alex-ju) [inline website/docs/components/dropdown/partials/code/component-api.md:20]: Following our naming convention, should this be just `[D]`?

File: website/docs/components/segmented-group/partials/code/component-api.md [hunk 5]
```diff
@@ -1,29 +1,79 @@
 ## Component API
 
+### SegmentedGroup
+
 <Doc::ComponentApi as |C|>
+  <C.Property @name="<[S].Button>" @type="yielded component">
```
Comment (alex-ju) [inline website/docs/components/segmented-group/partials/code/component-api.md:6]: `[SG]`?

File: website/docs/components/side-nav/partials/code/component-api.md [hunk 6]
```diff
@@ -217,29 +220,35 @@ The `SideNav::List` component is used to wrap and contain the `SideNav::List::Ti
...
   </C.Property>
 </Doc::ComponentApi>
 
-### SideNav::List::BackLink
+##### [L].BackLink
+
+The `SideNav::List::BackLink` component, yielded as contextual component.
 
-The `SideNav::List::BackLink` component uses the generic `Hds::Interactive` component. For more details about this utility component please refer to [its documentation page](/utilities/interactive).
+It internally uses the [`Hds::Interactive`](/utilities/interactive) utility component. For more details about this component API please refer to [its documentation page](/utilities/interactive?tab=code#component-api).
```
Comment (KristinLBradley) [inline website/docs/components/side-nav/partials/code/component-api.md:251]:
```suggestion
It internally uses the [`Hds::Interactive`](/utilities/interactive) utility component. For more details about this component API, please refer to [its documentation page](/utilities/interactive?tab=code#component-api).
```

File: website/docs/components/side-nav/partials/code/component-api.md [hunk 7]
```diff
@@ -262,22 +271,26 @@ The `SideNav::List::BackLink` component uses the generic `Hds::Interactive` comp
...
   </C.Property>
 </Doc::ComponentApi>
 
-### SideNav::List::Link
+##### [L].Link
+
+The `SideNav::List::Link` component, yielded as contextual component.
 
-The `SideNav::List::Link` component uses the generic `Hds::Interactive` component. For more details about this utility component please refer to [its documentation page](/utilities/interactive).
+It internally uses the [`Hds::Interactive`](/utilities/interactive) utility component. For more details about this component API please refer to [its documentation page](/utilities/interactive?tab=code#component-api).
```
Comment (KristinLBradley) [inline website/docs/components/side-nav/partials/code/component-api.md:293]:
```suggestion
It internally uses the [`Hds::Interactive`](/utilities/interactive) utility component. For more details about this component API, please refer to [its documentation page](/utilities/interactive?tab=code#component-api).
```

File: website/docs/components/side-nav/partials/code/component-api.md [hunk 8]
```diff
@@ -129,7 +131,9 @@ The `SideNav::Header::HomeLink` component uses the generic `Hds::Interactive` co
 
 ### SideNav::Header::IconButton
 
-The `SideNav::Header::IconButton` component uses the generic `Hds::Interactive` component. For more details about this utility component please refer to [its documentation page](/utilities/interactive).
+The `SideNav::Header::IconButton` component.
+
+It internally uses the [`Hds::Interactive`](/utilities/interactive) utility component. For more details about this component API please refer to [its documentation page](/utilities/interactive?tab=code#component-api).
```
Comment (KristinLBradley) [inline website/docs/components/side-nav/partials/code/component-api.md:136]:
```suggestion
It internally uses the [`Hds::Interactive`](/utilities/interactive) utility component. For more details about this component API, please refer to [its documentation page](/utilities/interactive?tab=code#component-api).
```

**Review by alex-ju — APPROVED** [review]
Reviewed the format and spotted a few inconsistencies in naming contextual components, otherwise it's great to see them aligned! 🙌 

I have some doubts about the `yield` being listed as any other argument, when in practice we're referring to the block content, but maybe users understand this correctly.

File: website/docs/components/accordion/partials/code/component-api.md [hunk 9]
```diff
@@ -17,18 +20,28 @@ The `Accordion` component serves as a wrapper to group one or more `Accordion::I
 
 ### Contextual Components
```
Comment (KristinLBradley) [inline website/docs/components/accordion/partials/code/component-api.md:21]:
```suggestion
### Contextual components
```

**Comment by KristinLBradley** [comment]
> Reviewed the format and spotted a few inconsistencies in naming contextual components, otherwise it's great to see them aligned! 🙌
> 
> I have some doubts about the `yield` being listed as any other argument, when in practice we're referring to the block content, but maybe users understand this correctly.

Listing the yield as an argument seems a bit strange to me as well.

**Review by KristinLBradley — APPROVED** [review]
I think this is a big improvement, great work!

I just left a few punctuation & similar suggestions.

File: website/docs/components/app-footer/partials/code/component-api.md (hunk 3 above)
Comment (didoo) [reply website/docs/components/app-footer/partials/code/component-api.md:12]: Ahh it's a leftover of previous test. I'm going to remove it.

**Review by zamoore — APPROVED** [review]
I really like the new doc format. It's much easier to read now. Great work.

File: website/docs/components/dropdown/partials/code/component-api.md (hunk 4 above)
Comment (didoo) [reply website/docs/components/dropdown/partials/code/component-api.md:20]: @alex-ju Yes, but unfortunately we documented it (in the how to use) as `<Hds::Dropdown as |dd|>` and now this `dd` [is all over the places](https://github.com/search?q=%22%3CHds%3A%3ADropdown+as%22++user%3Ahashicorp&type=code&ref=advsearch) :( 
So I tried to keep the naming in the "Component API" block consistent with what we have in the "how to use" block. More than happy to update the "how to use" block above, in this PR, if that's OK for you.

File: website/docs/components/segmented-group/partials/code/component-api.md (hunk 5 above)
Comment (didoo) [reply website/docs/components/segmented-group/partials/code/component-api.md:6]: Same as above, we've documented in the "How to use" as `<Hds::SegmentedGroup as |S|>`. Should I change it also there and use `SG` for both?

File: website/docs/components/dropdown/partials/code/component-api.md (hunk 4 above)
Comment (alex-ju) [reply website/docs/components/dropdown/partials/code/component-api.md:20]: ah, I see.. yeah, that's a good point. I guess consistent within the component is more of a priority. If not an overkill, I would like to make this change, but can do it in a follow up if you'd like to get this in sooner

Comment (didoo) [reply website/docs/components/dropdown/partials/code/component-api.md:20]: No, easy/cleaner to do in this context

Comment (didoo) [reply website/docs/components/dropdown/partials/code/component-api.md:20]: @alex-ju on a second thought, there were too many places where they should have been updated, for consistency, so I created a new PR: https://github.com/hashicorp/design-system/pull/2247

**Pull request merged**